core/src/scene/drawRule.cpp

```diff
@@ -174,7 +174,9 @@ void DrawRuleMergeSet::apply(const Feature& _feature, const SceneLayer& _layer,
         }
 
         bool visible;
-        if (rule.get(StyleParamKey::visible, visible) && !visible) {
+        if ((rule.get(StyleParamKey::visible, visible) && !visible) ||
+            (rule.get(StyleParamKey::outline_visible, visible) && !visible) ||
+            (rule.get(StyleParamKey::text_visible, visible) && !visible)) {
```

Comment:
This doesn't produce the correct behavior for these parameters. Under this logic, if a `lines` style has an `outline` node with `visible: false`, this will cause both the line and the outline to __not__ draw.